internal/api/core.go

```diff
@@ -110,6 +113,11 @@ func NewTokenValidator(provider *gophercloud.ProviderClient, eo gophercloud.Endp
 	return &tv, err
 }
 
+func (p *v1Provider) OverrideGenerateToken(generateTransferToken func() string) *v1Provider {
```

Comment:
I would like this naming to be more consistent:
```suggestion
func (p *v1Provider) OverrideGenerateTransferToken(generateTransferToken func() string) *v1Provider {
```